### PR TITLE
[Merged by Bors] - feat: port Data.PNat.Defs

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
First attempt - I hope it's going in the right direction

ported from [70fd9563a21e7b963887c9360bd29b2393e6225a](https://github.com/leanprover-community/mathlib/commit/70fd9563a21e7b963887c9360bd29b2393e6225a)

(technically from an earlier commit, but it hasn't changed since)

### CODE REVIEW

File: Mathlib/Data/Pnat/Defs.lean [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+This file contains the definitions, and basic results.
+Most algebraic facts are deferred to `data.pnat.basic`, as they need more imports.
+-/
+
+
+/-- `ℕ+` is the type of positive natural numbers. It is defined as a subtype,
+  and the VM representation of `ℕ+` is the same as `ℕ` because the proof
+  is not stored. -/
+def Pnat :=
```
Comment (digama0) [inline Mathlib/Data/Pnat/Defs.lean:21]:
```suggestion
def PNat :=
```
And in the file name and elsewhere...

File: Mathlib/Data/PNat/Defs.lean [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+This file contains the definitions, and basic results.
+Most algebraic facts are deferred to `Data.PNat.Basic`, as they need more imports.
+-/
+
+
+/-- `ℕ+` is the type of positive natural numbers. It is defined as a subtype,
+  and the VM representation of `ℕ+` is the same as `ℕ` because the proof
+  is not stored. -/
+def PNat :=
+  { n : ℕ // 0 < n }
```
Comment (semorrison) [inline Mathlib/Data/PNat/Defs.lean:22]:
```suggestion
def PNat := { n : ℕ // 0 < n }
```

File: Mathlib/Data/PNat/Defs.lean [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+-- Porting note: no `simp` due to eagerly elaborated coercions
+theorem mk_coe (n h) : ((⟨n, h⟩ : ℕ+) : ℕ) = n :=
+  rfl
+#align pnat.mk_coe PNat.mk_coe
+
+/-- Predecessor of a `ℕ+`, as a `ℕ`. -/
+def natPred (i : ℕ+) : ℕ :=
+  i - 1
+#align pnat.natPred PNat.natPred
```
Comment (semorrison) [inline Mathlib/Data/PNat/Defs.lean:49]:
```suggestion
#align pnat.nat_pred PNat.natPred
```
And below. Could you double check all the #aligns, then mark as ready for review again?

Comment (Ruben-VandeVelde) [reply Mathlib/Data/PNat/Defs.lean]: Doh. Thanks, done

File: Mathlib/Data/PNat/Defs.lean [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+/-- If `h : k | m`, then `k * (div_exact m k) = m`. Note that this is not equal to `m / k`. -/
+def divExact (m k : ℕ+) : ℕ+ :=
+  ⟨(div m k).succ, Nat.succ_pos _⟩
+#align pnat.div_exact PNat.divExact
+
+end PNat
+
+/-
+section CanLift
```
Comment (semorrison) [inline Mathlib/Data/PNat/Defs.lean:279]: Can we have a `-- Porting note:` explaining why these are commented out?

**Pull request closed without merging**